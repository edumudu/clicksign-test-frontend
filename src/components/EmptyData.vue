<template>
  <div :class="$style.emptyData">
    <EmptyDataSvg />

    <h3 :class="$style.title">
      <slot>
        Nenhum contato foi criando ainda.
      </slot>
    </h3>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

import EmptyDataSvg from '@/assets/svg/empty-data.svg?inline';

export default defineComponent({
  name: 'EmptyData',

  components: {
    EmptyDataSvg,
  },
});
</script>

<style lang="scss" module>
.emptyData {
  text-align: center;
}

.title {
  font-weight: 400;
  font-size: 1rem;
  color: var(--text-color);
  margin-top: 1.5rem;
}
</style>
