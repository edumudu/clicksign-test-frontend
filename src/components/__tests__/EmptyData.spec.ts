import { shallowMount } from '@vue/test-utils';

import EmptyData from '../EmptyData.vue';

describe('EmptyData.vue', () => {
  it('Should match snapshot', () => {
    const wrapper = shallowMount(EmptyData);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('Should render content passed in default slot', async () => {
    const wrapper = shallowMount(EmptyData, {
      slots: {
        default: '<div>Test</div>',
      },
    });

    expect(wrapper.text()).toBe('Test');
  });
});
