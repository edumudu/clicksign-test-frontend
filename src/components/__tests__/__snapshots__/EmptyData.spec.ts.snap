// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyData.vue Should match snapshot 1`] = `
<div
  class="emptyData"
>
  <empty-data-svg-stub />
  <h3
    class="title"
  >
    
     Nenhum contato foi criando ainda. 
    
  </h3>
</div>
`;
